### PR TITLE
Mail sender Full TLS

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22549

## Description
So far Mail Sender integration supported semi TLS connection (the first creation on the SERVER object was without SSL) such that some client's mailboxes failed to configured in such a way. Added an option for Full TLS including the creation of the SERVER object.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests: Working on getting our own instance to run test PB 
- [ ] Documentation

### CODE REVIEW

**Comment by ShahafBenYakir** [comment]
@glicht According to Sylvain Meras - smeras@paloaltonetworks.com (Don't think he has a Github user) who asked for this enhancement, there are some customers that needs Full TLS connection wheres ours is almost full. 

I added two different instances to https://dev3.demisto.works/#/settings/servers - Mail Sender Shahaf. 
The first supports regular TlS as we had so far where the second supports Full TLS and will fail for the regular TLS. You can go ahead and test that.

**Comment by glicht** [comment]
See my comments at: https://github.com/demisto/etc/issues/22549#issuecomment-605658573

**Comment by ShahafBenYakir** [comment]
@glicht Changed the implementation to be with drop-down instead of checkboxes.
The only possible issue is that for users that already configured the integration with TLS 
it will appear like this: 
Settings 2020-03-29 21-33-53

**Comment by glicht** [comment]
It looks ok. Please do an integration upgrade check with 5.0 too.

File: Integrations/MailSenderNew/MailSenderNew.yml [hunk 1]
```diff
@@ -25,11 +25,14 @@ configuration:
   name: fqdn
   required: false
   type: 0
-- defaultvalue: 'false'
-  display: Use TLS for connection
+- display: Use TLS for connection
```
Comment (glicht) [inline Integrations/MailSenderNew/MailSenderNew.yml]: Please use display name `Connection Security` with options:
* None
* STARTTLS
* SSL/TLS

By default None should be selected.

**Comment by ShahafBenYakir** [comment]
@glicht Should be fine now, have a look.
I did an integration update and will do the same on 5.0.

**Comment by ShahafBenYakir** [comment]
@kirbles19 Please go over the MD and YML. 
We will also need to add an explanation of the new dropdown configuration parameter in the integration docs which replaces the old one called `Use TLS for connection`.
Something like this should work:
### connection security: 
- `None` will have no connection security at all. 
- `SSL/TLS` first establishes a secure `SSL/TLS` connection and conducts SMTP over that connection.
- STARTTLS starts with unencrypted SMTP and then switches to SSL/TLS to ensure security.

File: Integrations/MailSenderNew/MailSenderNew.py [hunk 2]
```diff
@@ -333,19 +334,27 @@ def main():
...
-        SERVER = SMTP(demisto.getParam('host'), int(demisto.params().get('port', 0)), local_hostname=FQDN)
-        SERVER.ehlo()
+
         # TODO - support for non-valid certs
-        if demisto.getParam('tls'):
-            SERVER.starttls()
+        if TLS == 'SSL/TLS':
+            SERVER = SMTP_SSL(demisto.getParam('host'), int(demisto.params().get('port', 0)), local_hostname=FQDN)
+        else:
+            SERVER = SMTP(demisto.getParam('host'), int(demisto.params().get('port', 0)), local_hostname=FQDN)  # type: ignore
```
Comment (glicht) [inline Integrations/MailSenderNew/MailSenderNew.py:348]: When using `type: ignore` use the syntax: `# type: ignore[<error-name>]`  See: https://xsoar.pan.dev/docs/integrations/linting#mypy

File: Integrations/MailSenderNew/MailSenderNew.py [hunk 3]
```diff
@@ -333,19 +334,27 @@ def main():
...
+
+        SERVER.ehlo()   # type: ignore
```
Comment (glicht) [inline Integrations/MailSenderNew/MailSenderNew.py:350]: same remark regarding mypy ignore

File: Integrations/MailSenderNew/MailSenderNew.py [hunk 4]
```diff
@@ -333,19 +334,27 @@ def main():
...
+        # For BC purposes where TLS was a checkbox (no value only true or false) if TLS=True or TLS='STARTTLS' we enter
+        # this condition, otherwise it means TLS is not configured (TLS=False) or is set to 'SSL/TLS' or 'None'.
+        if TLS is True or TLS == 'STARTTLS':
+            SERVER.starttls()   # type: ignore
```
Comment (glicht) [inline Integrations/MailSenderNew/MailSenderNew.py:354]: fix mypy ignore

File: Integrations/MailSenderNew/MailSenderNew.py [hunk 5]
```diff
@@ -333,19 +334,27 @@ def main():
...
         user, password = get_user_pass()
         if user:
-            SERVER.login(user, password)
+            SERVER.login(user, password)    # type: ignore
```
Comment (glicht) [inline Integrations/MailSenderNew/MailSenderNew.py:357]: fix mypy ignore

File: Integrations/MailSenderNew/MailSenderNew.py [hunk 6]
```diff
@@ -360,8 +369,8 @@ def main():
             msg['Subject'] = 'Test mail from Demisto'
             msg['From'] = FROM
             msg['To'] = FROM
-            SERVER.sendmail(FROM, [FROM], msg.as_string())
-            SERVER.quit()
+            SERVER.sendmail(FROM, [FROM], msg.as_string())  # type: ignore
+            SERVER.quit()   # type: ignore
```
Comment (glicht) [inline Integrations/MailSenderNew/MailSenderNew.py:373]: fix mypy ignore

**Comment by kirbles19** [comment]
@ShahafBenYakir Done.

**Pull request merged**